subprojects/packagefiles/glfw/tests/meson.build

```diff
@@ -0,0 +1,226 @@
+###  GLFW test programs
+#@ Build check
+if get_option('build-tests') == false
+  subdir_done()
+endif
+
+#@ Constants
+tstdirs = include_directories('../include', '../src', '../deps')
+glad_gl = '../deps/glad_gl.c'
+glad_vulkan = '../deps/glad_vulkan.c'
+getopt = '../deps/getopt.c'
+tinycthread = '../deps/tinycthread.c'
+
+#@ Flags
+if sys_os == 'linux'
+  c_flags = ['-D_XOPEN_SOURCE=700']
+endif
+
+#@ Test executables
+tst_clipboard = executable('clipboard',
```

Comment:
Similar to what I mentioned by the examples... it might be easier to define all these in a foreach loop as they are mostly the same.